src/qml/WiFiPage.qml

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (C) 2023 - Arseniy Movshev <dodoradio@outlook.com>
+ *               2017-2022 - Chupligin Sergey <neochapay@gmail.com>
+ *               2021 - Darrel Griët <idanlcontact@gmail.com>
+ *               2016 - Sylvia van Os <iamsylvie@openmailbox.org>
+ *               2015 - Florent Revest <revestflo@gmail.com>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/* A large proportion of this code has been referenced from Nemomobile-UX Glacier-Settings, published at https://github.com/nemomobile-ux/glacier-settings/blob/master/src/plugins/wifi/WifiSettings.qml
+ */
+
+import QtQuick 2.9
+import org.asteroid.controls 1.0
+import org.asteroid.utils 1.0
+import Connman 0.2
+import QtQuick.VirtualKeyboard 2.4
+
+Item {
+    id: root
+    TechnologyModel {
+        id: wifiModel
+        name: "wifi"
+        onCountChanged: {
+            console.log("COUNT CHANGE " + count)
+        }
+        onScanRequestFinished: {
+            console.log("SCAN FINISH")
+        }
+    }
+
+    NetworkTechnology {
+        id: wifiStatus
+        path: "/net/connman/technology/wifi"
+        onPoweredChanged: {
+            console.log("POWER CHANGE ") + powered
+            if (powered)
+                wifiModel.requestScan()
+        }
+    }
+
+    ListView {
+        id: wifiList
+        model: wifiModel
+        width: parent.width*0.7
+        anchors.horizontalCenter: parent.horizontalCenter
+        height: parent.height
+        header: Item {
+            //this is literally a statuspage
```

Comment:
I haven't looked at the exact diff but any reason why you're not re-using/extending the existing statuspage instead ?
Not having had time to look into it yet is a valid question, I'm just curious

Reply:
ok, the comment here is old and misleading. Basically, this is/was the visual components of a statuspage, but they're then moved and resized to make way for the rest of the menu items when wifi is enabled. I'll add this to my cleanup todo list.